contracts/0.8.9/SelfOwnedStETHBurner.sol

```diff
@@ -0,0 +1,312 @@
+// SPDX-FileCopyrightText: 2021 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts-v4.4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4.4/token/ERC721/IERC721.sol";
+
+import "./interfaces/IStETH.sol";
+import "./interfaces/IBeaconReportReceiver.sol";
+
+/**
+  * @title Interface defining a Lido liquid staking pool
+  * @dev see also [Lido liquid staking pool core contract](https://docs.lido.fi/contracts/lido)
+  */
+interface ILido {
+    /**
+      * @notice Destroys given amount of shares from account's holdings,
+      * @param _account address of the shares holder
+      * @param _sharesAmount shares amount to burn
+      * @dev incurs stETH token rebase by decreasing the total amount of shares.
+      */
+    function burnShares(address _account, uint256 _sharesAmount) external returns (uint256 newTotalShares);
+
+    /**
+      * @notice Gets authorized oracle address
+      * @return address of oracle contract
+      */
+    function getOracle() external view returns (address);
+}
+
+/**
+  * @title Interface for the Lido Beacon Chain Oracle
+  */
+interface IOracle {
+    /**
+     * @notice Gets currently set beacon report receiver
+     * @return address of a beacon receiver
+     */
+    function getBeaconReportReceiver() external view returns (address);
+}
+
+/**
+  * @title A dedicated contract for enacting stETH burning requests
+  * @notice See the Lido improvement proposal #6 (LIP-6) spec.
+  * @author Eugene Mamin <TheDZhon@gmail.com>
+  *
+  * @dev Burning stETH means 'decrease total underlying shares amount to perform stETH token rebase'
+  */
+contract SelfOwnedStETHBurner is IBeaconReportReceiver {
```

Comment:
Not 100% happy with `SelfOwned` prefix, what do you mean by `SelfOwned`? Not governed? Maybe simplify to `StETHBurner`? And why not `SharePriceRecoverer`? Sounds more positive for me.

Reply:
SelfOwned prefix meaning applies to `StETH` and not `Burner`